### DISCUSSION TITLE
Who is responsible for server-side and front-end compatibility issues?[app can't connect]

### DISCUSSION THREAD

**Discussion by qcks** [immich-app/immich#13756] [Q&A] [answered]
### The bug

众所周知ios的app从商店更新后用户不能降级版本，android的虽然可以手动处理但是一般用户不会。服务端为了稳定性不应该频繁升级！那么二者各跑各的必将乱成一团。好歹也这么大的code行数了，这点小问题不好解决吗？
EN：
It is well known that ios apps can't be downgraded by the user after updating from the store, android ones can be handled manually but the average user won't. The server side should not be upgraded frequently for the sake of stability! Then both of them will be in a mess if they run their own way. At least it's such a big line of code, can't this small problem be solved?

The app releases a new version that is not compatible with the previous version of the service, forcing the service to be upgraded, what if my data is lost or messed up?
————————————————————————————
每个人的私人数据都不应该成为你肆意炫技的耗材

### The OS that Immich Server is running on

null

### Version of Immich Server

null

### Version of Immich Mobile App

null

### Platform with the issue

- [X] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
null
```

### Your .env content

```Shell
null
```

### Reproduction steps

null

### Relevant log output

```shell
null
```

**Comment by alextran1502** [comment, marked as answer]
As noted, the app is still in heavy development mode. When the app reach stable release, we will have better backward compatibility between version

You should also have backup of your data. Please follow back up and restore guide in our documentation page

**Reply by qcks** [reply]
http://192.168.1.10:2283/api/
app:v1.118.0
server:v1.118.3
version incompatibility

ApiException 400: Socket operation failed: GET /server/ping (Inner exception: ClientException with SocketException: Connection failed (OS Error: No route to host, errno = 65), address = 192.168.1.10, port = 2283, uri=http://192.168.1.10:2283/api/server/ping)

#0      IOClient.send (package:http/src/io_client.dart:154)
<asynchronous suspension>
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:93)
<asynchronous suspension>
#2      ApiClient.invokeAPI (package:openapi/api_client.dart:101)
<asynchronous suspension>
#3      ServerApi.pingServer (package:openapi/api/server_api.dart:492)
<asynchronous suspension>
#4      Future.timeout.<anonymous closure> (dart:async/future_impl.dart:963)
<asynchronous suspension>
#5      ApiService._isEndpointAvailable (package:immich_mobile/services/api.service.dart:106)
<asynchronous suspension>
#6      ApiService._resolveEndpoint (package:immich_mobile/services/api.service.dart:87)
<asynchronous suspension>
#7      ApiService.resolveAndSetEndpoint (package:immich_mobile/services/api.service.dart:69)
<asynchronous suspension>
#8      LoginForm.build.getServerLoginCredential (package:immich_mobile/widgets/forms/login/login_form.dart:104)
<asynchronous suspension>

**Reply by bo0tzz** [reply]
Those versions are compatible, you have a different issue, namely something being wrong with your network setup.

**Comment by qcks** [comment]
@alextran1502  Since there is no stable version, don't publish the app to the app market because the default behavior of the app market is to update the app to the latest version, my and even more people's private data is not a consumable for you to show off your skills.

**Comment by lj4o7** [comment]
ios用户：app又自动升级了，服务器端叒不匹配了，好不容易弄好的数据库叕废了，cao
大佬：苹果的问题，爱用用不用滚
